frontend/src/ComponentConfigNode.svelte

```diff
@@ -0,0 +1,199 @@
+    min-height: 100%;
+    display: flex;
+    flex-direction: column;
+    flex-wrap: nowrap;
+    justify-content: center;
+    align-items: stretch;
+    align-content: center;
+  }
+
+  input {
+    background: inherit;
+    color: inherit;
+    border: none;
+    border-bottom: 1px solid white;
+    flex-grow: 1;
+    margin-left: 1ch;
+  }
+
+  li {
+    margin-bottom: 1em;
+    display: flex;
+    flex-direction: row;
+    flex-wrap: nowrap;
+    justify-content: space-evenly;
+    align-items: baseline;
+    align-content: left;
+    white-space: nowrap;
+  }
+  input:focus {
+    outline: none;
+    box-shadow: inset 0 -1px 0 var(--main-fg-color);
+  }
+
+  /* Hide spinner buttons for numeric inputs
+     Source: https://stackoverflow.com/questions/3790935/can-i-hide-the-html5-number-input-s-spin-box/4298216#4298216 */
+  input::-webkit-outer-spin-button,
+  input::-webkit-inner-spin-button {
+    -webkit-appearance: none;
+    margin: 0;
+  }
+
+  input[type="number"] {
+    -moz-appearance: textfield;
+  }
+</style>
+
+<script>
+  import Checkbox from "./Checkbox.svelte";
+  export let node, element, optional;
+  let listElement, dictKey, dictValue, dataclassFields, unionTypeSelect;
+  $: element;
+  $: dataclassFields;
+  if (
+    node["type"] == "typing.List" ||
+    node["type"] == "typing.Tuple" ||
+    node["type"] == "typing.Dict"
+  ) {
+    element = [];
+  } else if (node["type"] == "typing.Union") {
+    unionTypeSelect = node["args"][0];
+  }
+  if (node["type"] == "ofrak_type.range.Range") {
+    element = [];
+  } else if (node["fields"] != null) {
+    element = {};
+  }
+  if (node["default"] != null) {
+    element = node["default"];
+  }
+  if (optional) {
+    element = null;
+  }
+
+  const addElementToArray = () => {
+    element = [...element, listElement];
+    element = element;
+    listElement = null;
+  };
+
+  const addElementToDict = () => {
+    element = [...element, [dictKey, dictValue]];
+    element = element;
+    dictKey = null;
+    dictValue = null;
+  };
+</script>
+
+<div class="container">
+  <div class="inputs">
+    {#if node["type"] == "typing.Optional"}
+      {#each node["args"] as arg}
+        <svelte:self node="{arg}" optional="true" bind:element="{element}" />
+      {/each}
+    {:else if node["type"] == "builtins.bool"}
+      <li>
+        <Checkbox bind:checked="{element}" />
+      </li>
+    {:else if node["type"] == "builtins.str"}
+      <li>
+        <input bind:value="{element}" />
+      </li>
+    {:else if node["type"] == "builtins.bytes"}
+      <li>
+        <input bind:value="{element}" />
+      </li>
+    {:else if node["type"] == "builtins.int"}
+      <li>
+        <input type="number" bind:value="{element}" />
+      </li>
+    {:else if node["type"] == "typing.List"}
+      <li>
+        {#each node["args"] as arg}
+          <svelte:self node="{arg}" bind:element="{listElement}" />
+          <button on:click="{addElementToArray}">Add Element</button>
```

Comment:
This is a good idea but not super straight forward to implement for reasons such as:
- displaying the value of the element when the value is another list or other larger structure
- we are not currently tracking items in objects, just kind of plopping them in there
- im sure others...
I think this will be easier to implement after I attack the CSS with @rbs-jacob and find a nice way of displaying the current config inside the view.

Reply:
@dannyp303 is this more doable now or still complicated?

Reply:
oh sorry didnt comment, but its done
<img width="815" alt="Screen Shot 2023-05-04 at 2 31 50 PM" src="https://user-images.githubusercontent.com/14964383/236296710-7ea2666c-4ae3-4620-bee4-47b9d7e326e8.png">